test/transcoding/cl-types.ll

```diff
@@ -19,9 +19,9 @@
 ;; }
 
 ; RUN: llvm-as %s -o %t.bc
-; RUN: llvm-spirv %t.bc -spirv-text -o %t.spv.txt
+; RUN: llvm-spirv -preserve-ocl-kernel-arg-type-metadata-through-string %t.bc -spirv-text -o %t.spv.txt
 ; RUN: FileCheck < %t.spv.txt %s --check-prefix=CHECK-SPIRV
-; RUN: llvm-spirv %t.bc -o %t.spv
+; RUN: llvm-spirv -preserve-ocl-kernel-arg-type-metadata-through-string %t.bc -o %t.spv
 ; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
 ; RUN: llvm-dis %t.rev.bc
 ; RUN: FileCheck < %t.rev.ll %s --check-prefix=CHECK-LLVM
```

Comment:
This seems to be a test for the preservation of metadata through round-trip translation from LLVM to SPIR-V, and then back to LLVM.  Preservation of this metadata cannot be preserved when implementing SPIR-V translation as per the SPIR-V specification.  As such, preservation of the metadata is not guaranteed through the (somewhat informal) principles of operation of the translator.

It should be made clearer in this test that it cannot be expected to work using SPIR-V as specified, i.e. without the `-preserve-ocl-kernel-arg-type-metadata-through-string` option.  This is essentially vendor-specific functionality, and not clearly stated as such (although it could also be useful functionality for testing purposes, and other special cases).

Assuming all behavior tested here is either vendor-specific or covered elsewhere, the filename should be changed to make it clear that this is testing nonstandard functionality.  Otherwise, parts of this test that would be relevant when not using the `-preserve-ocl-kernel-arg-type-metadata-through-string` option should be run both with and without that option.

Please consider whether a comment is needed at the start of the test.  (I expect that if the filename is updated, or if extra RUN lines are added with appropriate CHECK-PREFIXes to distinguish between standard and nonstandard behavior, then a comment should not be needed.)